tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc

```diff
@@ -0,0 +1,156 @@
+    TARGET_SPECIFIC_FLAGS += -D__DSP_PRESENT=1 -D__FPU_PRESENT=1
+    FLOAT=hard
+    TARGET_ARCH := cortex-m55
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m55+nodsp+nofp-generic))
+    CORE=M55
+    ARM_LDFLAGS := -Wl,--cpu=8.1-M.Main.mve.no_dsp.no_fp
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m55+nodsp+nofp
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m55+nofp-generic))
+    CORE=M55
+    ARM_LDFLAGS := -Wl,--cpu=8.1-M.Main.mve.no_fp
+    TARGET_SPECIFIC_FLAGS += -D__DSP_PRESENT=1
+    TARGET_ARCH := cortex-m55+nofp
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m33+nodsp-generic))
+    CORE=M33
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M33.no_dsp.no_fp
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m33+nodsp
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m33-generic))
+    CORE=M33
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M33
+    TARGET_SPECIFIC_FLAGS += -D__DSP_PRESENT=1 -D__FPU_PRESENT=1 -D__VTOR_PRESENT=1 -D__FPU_USED=1
+    FLOAT=hard
+    TARGET_ARCH := cortex-m33
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m0-generic))
+    CORE=M0
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M0
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m0
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m3-generic))
+    CORE=M3
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M3
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m3
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m4-generic))
+    CORE=M4
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M4.no_fp
+    TARGET_SPECIFIC_FLAGS+=
+    TARGET_ARCH := cortex-m4
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m4+fp-generic))
+    CORE=M4
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M4
+    TARGET_SPECIFIC_FLAGS+=
+    FLOAT=hard
+    TARGET_ARCH := cortex-m4+fp
+    GCC_TARGET_ARCH := cortex-m4
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m7+fp-generic))
+    CORE=M7
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M7
+    TARGET_SPECIFIC_FLAGS +=
+    FLOAT=hard
+    TARGET_ARCH := cortex-m7+fp
+    GCC_TARGET_ARCH :=  cortex-m7
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m7-generic))
+    CORE=M7
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M7.no_fp
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m7
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else
+    $(error "$(TARGET) not supported")
+  endif
+
+  # Toolchain specfic flags
+  ifeq ($(filter armclang,$(ALL_TAGS)),armclang)
+    CXX_TOOL  := armclang
+    CC_TOOL   := armclang
+    AR_TOOL   := armar
+    LD        := armlink
+    FLAGS_ARMC = \
+      --target=arm-arm-none-eabi \
+      -mcpu=$(TARGET_ARCH)
+    ifeq ($(BUILD_TYPE), release)
+       FLAGS_ARMC += -Wno-unused-private-field
+    endif
```

Comment:
I'm ok moving this to the top-level Makefile or leaving it here, whatever your preference. Would still like to use `TOOLCHAIN` instead of `TAGS`.

Reply:
Done